File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    public int[] solution(int brown, int yellow) {
        int[] answer = new int[2];
        
        for(int i=1; i<=yellow;i++){
            if(yellow%i!=0) continue;
            int w = i;
            int h = yellow/w;
            
            int brownCnt = (w+2)*2 + 2*h;
            
            if(brownCnt == brown){
                if(w+2>=h+2){
                    answer[0] = w+2;
                    answer[1] = h+2;
                }else{
                    answer[0] = h+2;
                    answer[1] = w+2;
                }

                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.kt
class Solution {
    fun solution(brown: Int, yellow: Int): IntArray {
        var answer = mutableListOf<Int>()
        
        var m = 0
        
        for(n in 1..yellow){
            if(yellow%n!=0) continue
            val m = yellow/n
            
            if((n+2)*(m+2)==(brown+yellow)){
                answer.add(n+2)
                answer.add(m+2)
                break
            }
        }
        
        answer.sortByDescending{it}
        
        
        return answer.toIntArray()
    }
}